File: angularapp/src/app/services/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Orders } from '../models/orders.model';
import { Observable } from 'rxjs';
import { APIURL } from '../constant/api_url';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private readonly http: HttpClient) { }


  placeOrder(order: Orders): Observable<Orders> {
    return this.http.post<Orders>(`${APIURL.APIurl}/orders`, order);
  }

  getAllOrders(): Observable<Orders[]> {
    return this.http.get<Orders[]>(`${APIURL.APIurl}/orders`);
  }

  getOrderById(orderId: number): Observable<Orders> {
    return this.http.get<Orders>(`${APIURL.APIurl}/orders/${orderId}`);
  }

  getAllOrdersByUserId(userId: number): Observable<Orders[]> {
    return this.http.get<Orders[]>(`${APIURL.APIurl}/orders/user/${userId}`);
  }

  updateOrder(orderId: number, orders: Orders): Observable<Orders> {
    return this.http.put<Orders>(`${APIURL.APIurl}/orders/${orderId}`, orders);
  }

  deleteOrder(orderId: number): Observable<void> {
    return this.http.delete<void>(`${APIURL.APIurl}/orders/${orderId}`);
  }
}

File: springapp/src/main/java/com/examly/springapp/service/OrderService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.model.Orders;

public interface OrderService {
    public Orders addOrder(Orders orders);
    public Orders getOrderById(int orderId);
    public List<Orders> getAllOrders();
    public Orders updateOrder(int orderId, Orders orderDetails);
    public boolean deleteOrder(int orderId);
    public List<Orders> getOrdersByUserId(int userId);
    public Orders createOrder(Orders orders);

}
